lib/_http_client.js

```diff
@@ -327,23 +327,33 @@ function socketCloseListener() {
 
   // NOTE: It's important to get parser here, because it could be freed by
   // the `socketOnData`.
-  var parser = socket.parser;
-  if (req.res && req.res.readable) {
+  const parser = socket.parser;
+  const res = req.res;
+  if (res) {
     // Socket closed before we emitted 'end' below.
-    if (!req.res.complete) req.res.emit('aborted');
-    var res = req.res;
-    res.on('end', function() {
+    if (!res.complete) {
```

Comment:
I'm not sure if the fact that it's no longer within `res.readable` is meaningful here or not. I just need to think this over quickly but the rest of this looks good.

Reply:
I've got a separate PR for that https://github.com/nodejs/node/pull/20230